### PR TITLE
Dynamic NFT support

### PR DESCRIPTION
**Pull request by tifrel**
Adds support for dynamic NFTs for as introduced [here](https://github.com/Mintbase/mb-contracts/pull/17).

### CODE REVIEW

File: packages/sdk/src/createMetadata/createMetadata.ts [hunk 1]
```diff
@@ -22,18 +22,19 @@ export const createMetadata = (
     mintersAllowlist = null,
     maxSupply = null,
     lastPossibleMint = null,
+    isDynamic = null,
     noMedia = false,
     noReference = false,
   } = args;
+  
+  if (contractAddress == null) {
```
Comment (sainthiago) [inline packages/sdk/src/createMetadata/createMetadata.ts]: !contractAddress ?

File: packages/sdk/src/lockMetadata/lockMetadata.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ */
+export const lockMetadata = (
+  args: LockMetadataArgs,
+): NearContractCall<LockMetadataArgsResponse> => {
+  const {
+    contractAddress = mbjs.keys.contractAddress,
+    metadataId,
+  } = args;
+
+  if (contractAddress == null) {
```
Comment (sainthiago) [inline packages/sdk/src/lockMetadata/lockMetadata.ts]: !contractAddress ?

File: packages/sdk/src/mintOnMetadata/mintOnMetadata.ts [hunk 3]
```diff
@@ -22,6 +22,10 @@ export const mintOnMetadata = (
     price,
   } = args;
 
+  if (contractAddress == null) {
```
Comment (sainthiago) [inline packages/sdk/src/mintOnMetadata/mintOnMetadata.ts]: !contractAddress ?

File: packages/sdk/src/updateMetadata/updateMetadata.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+): NearContractCall<UpdateMetadataArgsResponse> => {
+  const {
+    contractAddress = mbjs.keys.contractAddress,
+    metadataId,
+    metadata,
+    noReference = false,
+    noMedia = false,
+  } = args;
+
+  if (contractAddress == null) {
```
Comment (sainthiago) [inline packages/sdk/src/updateMetadata/updateMetadata.ts]: !contractAddress ?

**Review by rubenmarcus — APPROVED** [review]
shouldnt we specify the network on the methods?

**Review by sainthiago — APPROVED** [review]

**Pull request merged**